lib/src/router/normalized_path.dart

```diff
@@ -60,8 +66,14 @@ class NormalizedPath {
   ///
   /// The [start] parameter specifies the starting segment index (inclusive).
   /// The optional [end] parameter specifies the ending segment index (exclusive).
-  NormalizedPath subPath(final int start, [final int? end]) =>
-      NormalizedPath._(segments.sublist(start, end));
+  NormalizedPath subPath(final int start, [int? end]) {
+    end ??= length;
+    if (start == end) return NormalizedPath.empty;
+    if (start == 0 && end == length) {
+      return this; // since NormalizedPath is immutable
```

Comment:
Suggestion: Maybe use `List.unmodifiable(result)` in `_normalize`?

Just to be safe for future changes and external behavior. Right now everything _inside_ this calls seems okay (as we always create our own `List` from the `path` `String`). 

But `segments` is still publicly exposed…

Reply:
.. if it doesn't hurt the benchmark too much 😃

Reply:
Not sure what happens in
```dart
external factory List.unmodifiable(Iterable elements);
```
but that was a lot less costly than I feared.

Reply:
Done